src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -1190,11 +1191,15 @@ void IMergeTreeDataPart::loadColumns(bool require)
         auto in = metadata_manager->read("columns.txt");
         loaded_columns.readText(*in);
 
-        for (const auto & column : loaded_columns)
+        for (auto & column : loaded_columns)
         {
-            const auto * aggregate_function_data_type = typeid_cast<const DataTypeAggregateFunction *>(column.type.get());
-            if (aggregate_function_data_type && aggregate_function_data_type->isVersioned())
-                aggregate_function_data_type->setVersion(0, /* if_empty */true);
+            auto callback = [](DataTypePtr & type)
```

Comment:
Maybe move such calls of `callOnNestedSimpleTypes` to separate functions? It looks like copy-paste.